spec-tests/src/test/java/org/candlepin/spec/imports/ImportUpdateSinglePoolTest.java

```diff
@@ -0,0 +1,100 @@
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.spec.imports;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.candlepin.dto.api.client.v1.AsyncJobStatusDTO;
+import org.candlepin.dto.api.client.v1.ConsumerDTO;
+import org.candlepin.dto.api.client.v1.EntitlementDTO;
+import org.candlepin.dto.api.client.v1.OwnerDTO;
+import org.candlepin.dto.api.client.v1.ProductDTO;
+import org.candlepin.dto.api.client.v1.SubscriptionDTO;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.data.builder.Export;
+import org.candlepin.spec.bootstrap.data.builder.ExportGenerator;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.builder.Products;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+@SpecTest
+public class ImportUpdateSinglePoolTest {
+
+    @Test
+    public void shouldBeAbleToMaintainMultipleImportedEntitlementsFromTheSamePool() throws Exception {
+        ApiClient adminClient = ApiClients.admin();
+        ProductDTO prod = Products.random();
+        ExportGenerator exportGenerator = new ExportGenerator(adminClient)
+            .minimal()
+            .withProduct(prod);
+        Export export = exportGenerator.export();
+        ConsumerDTO consumer = export.consumer();
+
+        List<EntitlementDTO> ents = adminClient.consumers().listEntitlements(consumer.getUuid());
+        assertThat(ents).singleElement();
+        EntitlementDTO ent1 = ents.get(0);
+
+        OwnerDTO owner = adminClient.owners().createOwner(Owners.random());
+        String ownerKey = owner.getKey();
+
+        AsyncJobStatusDTO importJob = adminClient.owners().importManifestAsync(ownerKey, List.of(), export.file());
+        importJob = adminClient.jobs().waitForJob(importJob);
+        assertEquals("FINISHED", importJob.getState());
+
+        List<SubscriptionDTO> subs = adminClient.owners().getOwnerSubscriptions(ownerKey);
+        assertThat(subs)
+            .singleElement()
+            .extracting(SubscriptionDTO::getProduct)
+            .isNotNull()
+            .returns(prod.getId(), ProductDTO::getId);
+
+        int expectedQuantity = 5;
+        JsonNode ent2 = adminClient.consumers()
+            .bindPool(consumer.getUuid(), subs.get(0).getUpstreamPoolId(), expectedQuantity)
+            .get(0);
+        Thread.sleep(1000);
```

Comment:
I tested this on myself(in my PR with bind) so we wouldn't waste so much time sleeping. I got to 450 ms, but to be safe we can use 500 and save a second in this case.

Tested with this annotation.
```
    @RepeatedTest(30)
    @Execution(ExecutionMode.SAME_THREAD)
```

I try it in this PR too and it is working with 500.

Reply:
Updated